catalog/validator/catalog-validator-metacardduplication/src/main/java/org/codice/ddf/validator/metacard/duplication/DuplicationValidator.java

```diff
@@ -51,18 +57,21 @@
 import org.slf4j.LoggerFactory;
 
 public class DuplicationValidator
-    implements MetacardValidator,
-        ReportingMetacardValidator,
+    implements PreIngestPlugin,
```

Comment:
Why did we move away from using the Validation API?

Reply:
`MetacardValidator` is not able to remove metacards from a `CreateRequest`, which we do to reject ingests

Reply:
Since this was a MetacardValidator before, we could reject duplicate attributes with enforce errors or warnings in the MetacardValidityMarkerPlugin, I don't think this is the case anymore since the MetacardValidityMarkerPlugin and now this plugin are both run at PreIngest.

Reply:
To me, it seems like rejecting and adding warnings/errors are separate cases, as well as separate from enforcing errors/warnings. We would want to reject before adding any errors or warnings (if configured to do so) to save time and effort. Doing that with the MetacardValidator would mean we need to throw an exception that would bubble up across many classes where it didn't necessarily need to be thrown.

Reply:
@peterhuffer does have a point though. Maybe we don't need to do this whole refactor and instead allow users to just reject errors or warnings using the `MarkerPlugin`. It'd certainly be less code changes. I guess we need to decide if rejecting separate from errors/warnings is a separate and worthy enough case instead of rejecting by errors/warnings